### PR TITLE
EASY-2746: Add column 'urn' and GET request

### PR DESCRIPTION
**Pull request by vesaakerman**
Fixes EASY-2746

#### When applied...
* it will add `URN` into the bag-index
* it will make `DOI` optional
* bags can be searched with a `DOI` or a `URN`

#### Related pull requests on github

repo                       | PR
-------------------------- | -----------------
easy-                      | [PR#](PRlink)

### CODE REVIEW

File: docs/api/api.yml [hunk 1]
```diff
@@ -148,6 +172,13 @@ components:
         type: string
       required: true
       description: digital object identifier of the targeted bag
+    URN:
+      in: path
+      name: URN
+      schema:
+        type: string
+      required: true
+      description: digital object identifier of the targeted bag
```
Comment (lindareijnhoudt) [inline docs/api/api.yml:181]:
```suggestion
      description: Uniform Resource Name of the targeted bag
```

File: src/test/scala/nl/knaw/dans/easy/bagindex/components/DatabaseSpec.scala [hunk 2]
```diff
@@ -35,9 +35,10 @@ class DatabaseSpec extends TestSupportFixture with BagIndexDatabaseFixture with
       DateTime.now()
     )
     val dois = List("10.5072/dans-x6f-kf6x", "10.5072/dans-x6f-kf66", "10.5072/dans-y7g-lg77")
+    val urns = List("urn:nbn:nl:ui:13-00-1haq", "urn:nbn:nl:ui:13-00-2haq", "urn:nbn:nl:ui:13-00-3haq")
 
-    (bagIds, times, dois).zipped.toList
-      .map { case (bagId, time, doi) => database.addBagInfo(bagId, baseId, time, doi) }
+    (bagIds zip times zip dois zip urns)
```
Comment (lindareijnhoudt) [inline src/test/scala/nl/knaw/dans/easy/bagindex/components/DatabaseSpec.scala:40]: what does this mean? It is repeated later also

File: src/test/scala/nl/knaw/dans/easy/bagindex/components/DatabaseSpec.scala [hunk 3]
```diff
@@ -119,37 +123,74 @@ class DatabaseSpec extends TestSupportFixture with BagIndexDatabaseFixture with
...
+    }
+  }
+
+  it should "return all bags with a certain URN (identifierType 'urn')" in {
+    val bagIds @ bagId1 :: bagId2 :: Nil = List.fill(2)(UUID.randomUUID())
+    val times @ time1 :: time2 :: Nil = List(
+      DateTime.parse("1992-07-30T16:00:00"),
+      DateTime.now()
+    )
+    val urns @ urn1 :: _ = List.fill(2)("urn:nbn:nl:ui:13-00-1haq")
```
Comment (lindareijnhoudt) [inline src/test/scala/nl/knaw/dans/easy/bagindex/components/DatabaseSpec.scala:150]: shouldn't the URN be unique?

Comment (vesaakerman) [reply src/test/scala/nl/knaw/dans/easy/bagindex/components/DatabaseSpec.scala:150]: In this test there are two bagIds and both of them have the same URN

File: src/test/scala/nl/knaw/dans/easy/bagindex/components/DatabaseSpec.scala (hunk 2 above)
Comment (vesaakerman) [reply src/test/scala/nl/knaw/dans/easy/bagindex/components/DatabaseSpec.scala:40]: The method `zipped` works only up to three elements.  Now we have four, that is why I have to use zip...zip...zip

**Pull request merged**